src/backend/executor/execMain.c

```diff
@@ -4091,6 +4091,7 @@ EvalPlanQualSetTuple(EPQState *epqstate, Index rti, HeapTuple tuple)
 {
 	EState	   *estate = epqstate->estate;
 
+	Assert(estate);
```

Comment:
how about instead it change code https://github.com/arenadata/gpdb/blob/fd5939d4f491238bbd28be88b98949cd0e2d6fe0/src/backend/executor/nodeLockRows.c#L84-L86 to
```c
 /* clear any leftover test tuple for this rel */ 
 Assert(node->lr_epqstate.estate) 
 EvalPlanQualSetTuple(&node->lr_epqstate, erm->rti, NULL);
```

Reply:
There's no error in this function if estate is null, but `EvalPlanQualSetTuple` itself requires it to be non-null. Moreover, it's better to assert in one place than look for all the places where `EvalPlanQualSetTuple` was called

Reply:
try use Insist

Reply:
Done